modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -1624,19 +1630,31 @@ bool PngEncoder::writeanimation(const Animation& animation, const std::vector<in
 
         for (j = 0; j < 6; j++)
             op[j].valid = 0;
-        deflateRectOp(frames[0].getPixels(), x0, y0, w0, h0, bpp, rowbytes, zbuf_size, 0);
-        deflateRectFin(zbuf.data(), &zsize, bpp, rowbytes, rows.data(), zbuf_size, 0);
 
-        if (first)
+        if (!animation.still_image.empty() && num_frames > 1)
         {
-            writeIDATs(m_f, 0, zbuf.data(), zsize, idat_size);
-            for (j = 0; j < 6; j++)
-                op[j].valid = 0;
-            deflateRectOp(frames[1].getPixels(), x0, y0, w0, h0, bpp, rowbytes, zbuf_size, 0);
+            CV_Assert(animation.still_image.type() == animation.frames[0].type() && animation.still_image.size() == animation.frames[0].size());
+            APNGFrame apngFrame;
+            Mat tmp;
+            if (animation.still_image.depth() == CV_16U)
+            {
+                animation.still_image.convertTo(tmp, CV_8U, 1.0 / 255);
+            }
```

Comment:
Most probably the conversion is not needed, if the animation patch in 16-bit too.

Reply:
currently APNG encoder converts 16U Mats to 8U before saving. no support for writing 16 bit APNG yet.

Reply:
Added issue to our tracker to make the limitation more obvious.